wgengine/wgtun/tun.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package wgtun
+
+import (
+	"errors"
+	"os"
+	"sync"
+
+	"github.com/tailscale/wireguard-go/device"
+	"github.com/tailscale/wireguard-go/tun"
+	"tailscale.com/types/logger"
+	"tailscale.com/wgengine/filter"
+)
+
+const MaxPacketSize = device.MaxContentSize
+
+var (
+	ErrFiltered     = errors.New("packet dropped by filter")
+	ErrPacketTooBig = errors.New("packet too big")
+)
+
+// Device wraps a tun.Device from wireguard-go,
+// augmenting it with filtering and packet injection.
+// All the added work happens in Read and Write:
+// the other methods delegate to the underlying innerDevice.
+type Device struct {
+	innerDevice tun.Device
+
+	// Inbound packets can be written directly into the TUN device,
```

Comment:
// outbound is ...

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

But I'm a little confused about directions here. "inbound packets can be written" sounds weird to me. I feel like you write outbound bounds and read inbound packets. Clarify somehow?

Reply:
Yes, sorry, the inbound/outbound has a different reference point here that matches the perspective of `wireguard-go` where `tun.Device` comes from. I've clarified this.